index.js

```diff
@@ -1,20 +1,32 @@
 'use strict';
-module.exports = (condition, interval) => new Promise((resolve, reject) => {
-	interval = typeof interval === 'number' ? interval : 20;
+const pTimeout = require('p-timeout');
 
-	const check = () => {
-		Promise.resolve().then(condition).then(val => {
-			if (typeof val !== 'boolean') {
-				throw new TypeError('Expected condition to return a boolean');
-			}
+module.exports = (condition, interval, opts) => {
```

Comment:
Probably makes sense to make `interval` part of `opts` as well

Reply:
I mentioned that above, it would be a breaking change, so wanted to check first. 

100% agree it's cleaner though

Reply:
I'm ok with that